Tests/Unit/Domain/Site/SiteHashServiceTest.php

```diff
@@ -0,0 +1,89 @@
+ *  (at your option) any later version.
+ *
+ *  The GNU General Public License can be found at
+ *  http://www.gnu.org/copyleft/gpl.html.
+ *
+ *  This script is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+use ApacheSolrForTypo3\Solr\Domain\Site\SiteHashService;
+use ApacheSolrForTypo3\Solr\Site;
+use ApacheSolrForTypo3\Solr\Tests\Unit\UnitTest;
+
+/**
+ * Testcase to check if the SiteHashService class works as expected.
+ *
+ * The unit test is used to make sure that the SiteHashService works as expected when the calls to Site:: are mocked
+ *
+ * @author Timo Hund <timo.hund@dkd.de>
+ */
+class SiteHashServiceTest extends UnitTest
+{
+
+    /**
+     * @return array
+     */
+    public function canResolveSiteHashAllowedSitesDataProvider() {
+        return [
+            'siteHashDisabled' => ['*', '*'],
+            'allSitesInSystem' => ['__all', 'solrtesta.local,solrtestb.local'],
+            'currentSiteOnly' => ['__current_site', 'solrtesta.local']
+        ];
+    }
+
+    /**
+     * @dataProvider canResolveSiteHashAllowedSitesDataProvider
+     * @test
+     */
+    public function canResolveSiteHashAllowedSites($allowedSitesConfiguration , $expectedAllowedSites)
+    {
+        $siteA = $this->getDumbMock(Site::class);
+        $siteA->expects($this->any())->method('getDomain')->will($this->returnValue('solrtesta.local'));
+        $siteB = $this->getDumbMock(Site::class);
+        $siteB->expects($this->any())->method('getDomain')->will($this->returnValue('solrtestb.local'));
+        $allSites = [$siteA, $siteB];
+
+            /** @var $siteHashServiceMock SiteHashService */
+        $siteHashServiceMock = $this->getMockBuilder(SiteHashService::class)->setMethods(['getAvailableSites','getSiteByPageId'])->getMock();
+        $siteHashServiceMock->expects($this->any())->method('getAvailableSites')->will($this->returnValue($allSites));
+        $siteHashServiceMock->expects($this->any())->method('getSiteByPageId')->will($this->returnValue($siteA));
+
+        $allowedSites = $siteHashServiceMock->getAllowedSitesForPageIdAndAllowedSitesConfiguration(1, $allowedSitesConfiguration);
+        $this->assertSame($expectedAllowedSites, $allowedSites, 'resolveSiteHashAllowedSites did not return expected allowed sites');
+    }
+
+    /**
+     * @test
+     */
+    public function getSiteHashForDomain()
+    {
+        $oldKey = $GLOBALS['TYPO3_CONF_VARS']['SYS']['encryptionKey'];
+        $GLOBALS['TYPO3_CONF_VARS']['SYS']['encryptionKey'] = 'testKey';
+
+        $service = new SiteHashService();
+        $hash1 = $service->getSiteHashForDomain('www.test.de');
```

Comment:
Please use `www.example.com` for such use-cases. `test.de` is Stiftung Warentest.